daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/Ref.scala

```diff
@@ -78,6 +78,53 @@ object Ref {
   type ParticipantId = IdString.ParticipantId
   val ParticipantId: IdString.ParticipantId.type = IdString.ParticipantId
 
+  sealed trait IdentityProviderId {
```

Comment:
I have noticed it is also needed in the daml-script.... well, in that case it needs to stay where it is.

Reply:
I am not complelty against adding those concept in LF. 
However, I do not sure script/trigger are a good reason to do so, 
I mean they depends on some ledger modules, so it could maybe be defined there.

For instance script import all those modules:  
- "//ledger-api/rs-grpc-bridge",
- "//ledger-service/cli-opts",
- "//ledger-service/jwt",
- "//ledger-service/lf-value-json",
- "//ledger/ledger-api-auth",
- "//ledger/ledger-api-client",
- "//ledger/ledger-api-common",
- "//ledger/participant-local-store",

Reply:
I moved out `IdentityProviderId` from daml-lf into `ledger-api-domain` and I believe it is actually a better place